src/generators/AbstractGenerator.ts

```diff
@@ -9,35 +9,34 @@ export interface CommonGeneratorOptions<P extends Preset = Preset> {
   };
   defaultPreset?: P;
   presets?: Presets<P>;
-  processorOptions?: ProcessorOptions;
+  processorOptions?: ProcessorOptions
 }
 
 export const defaultGeneratorOptions: CommonGeneratorOptions = {
   indentation: {
     type: IndentationTypes.SPACES,
     size: 2,
-  },
+  }
 };
 
 /**
  * Abstract generator which must be implemented by each language
  */
-export abstract class AbstractGenerator<Options extends CommonGeneratorOptions = CommonGeneratorOptions, RenderCompleteModelOptions = any> {
-  protected options: Options;
-  
+export abstract class AbstractGenerator<
+  Options extends CommonGeneratorOptions = CommonGeneratorOptions, 
+  RenderCompleteModelOptions = any> {
   constructor(
     public readonly languageName: string,
-    defaultOptions?: Options,
-    passedOptions?: Options,
-  ) {
-    this.options = this.mergeOptions(defaultOptions, passedOptions);
-  }
+    public readonly options: Options
+  ) { }
 
-  public abstract render(model: CommonModel, inputModel: CommonInputModel): Promise<RenderOutput>;
-  public abstract renderCompleteModel(model: CommonModel, inputModel: CommonInputModel, options: RenderCompleteModelOptions): Promise<RenderOutput>;
+  public abstract render(model: MetaModel, inputModel: InputMetaModel): Promise<RenderOutput>;
+  public abstract renderCompleteModel(model: MetaModel, inputModel: InputMetaModel, options: RenderCompleteModelOptions): Promise<RenderOutput>;
+  public abstract constrainToMetaModel(model: MetaModel): ConstrainedMetaModel;
+  public abstract splitMetaModel(model: MetaModel): MetaModel[];
 
-  public process(input: Record<string, unknown>): Promise<CommonInputModel> {
-    return InputProcessor.processor.process(input, this.options.processorOptions);
+  public async process(input: Record<string, unknown>): Promise<InputMetaModel> {
+    return await InputProcessor.processor.process(input, this.options.processorOptions);
```

Comment:
You don't need to wrap async with await, if return type of simple function (like above) is promise.

Reply:
True, removed 👍